src/App.tsx

```diff
@@ -15,48 +17,129 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface Good {
+  name: string;
+  length: number;
+  id: string;
+}
+
+interface SortType {
+  sortType: SortValue | null;
+  reverse: boolean;
+}
+
+enum SortValue {
+  Name = 'name',
+  Length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortType, reverse }: SortType,
+): Good[] {
+  let preparedGoods = goods.map((good) => ({
+    name: good,
+    length: good.length,
+    id: uuidv4(),
+  }));
+
+  if (sortType) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortValue.Name:
+          return good1[sortType].localeCompare(good2[sortType]);
+
+        case SortValue.Length:
+          return good1[sortType] - good2[sortType];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>({
+    sortType: null,
```

Comment:
Use an empty string for example and add it to your `enum` and separate reverse to another state with boolean value

Reply:
Also don't forget about Keep it simple principle and try to simplify the code as much as possible